lib/fs.js

```diff
@@ -509,15 +513,19 @@ function readFileAfterClose(err) {
   if (context.err || err)
     return callback(context.err || err);
 
-  if (context.size === 0)
-    buffer = Buffer.concat(context.buffers, context.pos);
-  else if (context.pos < context.size)
-    buffer = context.buffer.slice(0, context.pos);
-  else
-    buffer = context.buffer;
+  try {
+    if (context.size === 0)
+      buffer = Buffer.concat(context.buffers, context.pos);
+    else if (context.pos < context.size)
+      buffer = context.buffer.slice(0, context.pos);
+    else
+      buffer = context.buffer;
 
-  if (context.encoding) {
-    return tryToString(buffer, context.encoding, callback);
+    if (context.encoding) {
+      return tryToString(buffer, context.encoding, callback);
```

Comment:
`retryToString()` should be moved outside the try/catch block, it has a try/catch block of its own.

Reply:
right, thanks for pointing this out – I’ve just eliminated `tryToString` altogether